extensions/librdkafka/PublishKafka.cpp

```diff
@@ -187,81 +239,77 @@ bool PublishKafka::configureNewConnection(const std::shared_ptr<KafkaConnection>
   auto key = conn->getKey();
 
   if (key->brokers_.empty()) {
-    logger_->log_error("There are no brokers");
-    return false;
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "There are no brokers");
   }
   result = rd_kafka_conf_set(conf_, "bootstrap.servers", key->brokers_.c_str(), errstr.data(), errstr.size());
   logger_->log_debug("PublishKafka: bootstrap.servers [%s]", key->brokers_);
   if (result != RD_KAFKA_CONF_OK) {
-    logger_->log_error("PublishKafka: configure error result [%s]", errstr.data());
-    return false;
+    auto error_msg = utils::StringUtils::join_pack("PublishKafka: configure error result [%s]", errstr.data());
```

Comment:
I don't think it's gonna work as you expect. It just joins them, doesn't work as a format string. 

What you mean is more like:
```
auto error_msg = utils::StringUtils::join_pack("PublishKafka: configure error result [", errstr.data(), "]");
```

Reply:
Maybe we could use the formatting library used by spdlog to create these strings? I've mentioned directly depending on the formatting library in [MINIFICPP-1121](https://issues.apache.org/jira/projects/MINIFICPP/issues/MINIFICPP-1121).

Reply:
I actually did not expect I forgot this one :), @arpadboda what do you think about adding formatting library?

Reply:
I'm fine with that, but please make sure you do that in a separate commit and in case we start using that directly we should add the cmake dependency directly as well. 
https://github.com/apache/nifi-minifi-cpp/blob/master/ThirdParties.md

Reply:
okay, then I will do a follow-up refactoring later